contracts/test/utils/Permit.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts/cryptography/ECDSA.sol";
+import "forge-std/Test.sol";
+import "contracts/Spender.sol";
+import "contracts/utils/SpenderLibEIP712.sol";
+import "contracts/utils/SignatureValidator.sol";
+import { getEIP712Hash } from "contracts-test/utils/Sig.sol";
+
+contract Permit is Test {
+    function signSpendWithPermit(
+        uint256 privateKey,
+        SpenderLibEIP712.SpendWithPermit memory spendWithPermit,
+        bytes32 domainSeparator,
+        SignatureValidator.SignatureType sigType
+    ) internal returns (bytes memory sig) {
+        uint256 SPEND_WITH_PERMIT_TYPEHASH = 0x52718c957261b99fd72e63478d85d1267cdc812e8249f5a2623566c1818e1ed0;
+        bytes32 structHash = keccak256(
+            abi.encode(
+                SPEND_WITH_PERMIT_TYPEHASH,
+                spendWithPermit.tokenAddr,
+                spendWithPermit.requester,
+                spendWithPermit.user,
+                spendWithPermit.recipient,
+                spendWithPermit.amount,
+                spendWithPermit.actionHash,
+                spendWithPermit.expiry
+            )
+        );
+        bytes32 spendWithPermitHash = getEIP712Hash(domainSeparator, structHash);
+        if (sigType == SignatureValidator.SignatureType.EIP712 || sigType == SignatureValidator.SignatureType.WalletBytes32) {
```

Comment:
What's the reason to include `WalletBytes32` here?

Reply:
Some Foundry Test need it, like this:
[https://github.com/consenlabs/tokenlon-contracts/blob/master/contracts/test/RFQ.t.sol#L275](https://github.com/consenlabs/tokenlon-contracts/blob/master/contracts/test/RFQ.t.sol#L275)